timely_beliefs/tests/test_belief_utils.py

```diff
@@ -11,6 +12,16 @@
 from timely_beliefs.tests.utils import equal_lists
 
 
+def test_propagate_metadata_on_empty_frame():
+    df = BeliefsDataFrame(sensor=Sensor("test", unit="kW"))
+    df = df.for_each_belief(get_median_belief)
```

Comment:
I don't get why we compute something per belief (and later group as well) on an empty dataframe?

Reply:
To test whether calling these functions on an empty frame leads to losing the frame's metadata.